dvc/repo/metrics/show.py

```diff
@@ -85,12 +85,14 @@ def _read_metrics(repo, metrics, rev, onerror=None):
 
     res = {}
     for metric in metrics:
+        rel_metric_path = os.path.join(relpath, *fs.path.parts(metric))
```

Comment:
fs.path.join? As I recall one of the problems is windows behavior. dvcfs utlizizes posixpath.

Reply:
Um. Some unit tests are failing.
I am not sure if this would actually be a breaking change for windows 🤔 (and potentially vscode?) as we would be changing the returned path format

Reply:
My bad, metrics results are relying on os specific paths. I wonder if that won't be problematic in the future. In plots we are heavily relying on fs.path. This is inconsistent, we might need to consider unifying it at some point.